Crashlytics android

Redth opened the pull request.

Commit:
Bindings done.

Commit:
Devtools and samples added.

Commit:
Sample added.

Commit:
New sample fixed, Crashlytics fixed.

Commit:
Fixes and updates added.

Commit:
Project files fixed and cake file updated.

Commit:
Nuget path modified.

Redth committed:
Merge pull request #428 from adlair/Crashlytics

[Crashlytics] Bindings done.

Redth committed:
Rename assembly / package id’s

Redth committed:
Switch away from buildspec

Also extract .aar contents on externals

Redth committed:
build.cake improvements

- extract .aar contents
- stop using buildspec and just use msbuild
- nuget target for packing nugets instead of needing to pack on each build and copy to output
- update .csproj and .targets files to version in build script (so one source of truth) with xmlpoke

Redth committed:
Added targets files for XBD

We can’t embed the .aar files due to licensing so we use XamarinBuildDownload to grab them at app compile time instead.

Redth committed:
Binding fixes

Redth committed:
Remove ‘devtools’ addition

1. This binding was old and really from the _old_ fabric days, it doesn’t work with the newer versions properly.
2. We can merge some of the functionality this helped with into the main Crashlytics binding

Redth committed:
Merge samples into one

Redth committed:
Set msbuild props inside targets

`$(XamarinBuildDownloadDir)` wasn’t resolving when the properties resolved, so move the properties all into the target itself so they run later when the download dir property will be set.

Redth committed:
Add api key template to the sample

Redth committed:
Remove artwork from unknown origin

Redth committed:
Fix dependencies (add firebase core)

Redth committed:
Add some more comments

Redth committed:
Fix sample app manifest

Redth committed:
Add required resources to sample

Redth committed:
Fix layout / add readme

Redth committed:
Cleaned up some info

These keys were from a since deleted sample project, so there’s no risk of them in the history here, but removing them just for good measure.

Redth committed:
Add thirdpartynotice file to nupkgs

Redth committed:
Fix up external dep info license

Redth committed:
Fix xmlpoke of version to correct xpath

newky2k commented on Android/Crashlytics/source/CrashlyticsAnswers/CrashlyticsAnswers.csproj:
```diff
@@ -0,0 +1,67 @@
+        <Summary>Xamarin.Android bindings for Crashlytics Android - Answers</Summary>
+        <Description>Xamarin.Android bindings for Crashlytics Android - Answers</Description>
+        <PackageTags>Xamarin Android Xamarin.Android Crashlytics</PackageTags>
+        <Authors>Xamarin Inc.</Authors>
```
Needs to be Microsoft as the author

newky2k commented on Android/Crashlytics/source/CrashlyticsAnswers/CrashlyticsAnswers.csproj:
```diff
@@ -0,0 +1,67 @@
+        <Description>Xamarin.Android bindings for Crashlytics Android - Answers</Description>
+        <PackageTags>Xamarin Android Xamarin.Android Crashlytics</PackageTags>
+        <Authors>Xamarin Inc.</Authors>
+        <Owners>Xamarin Inc.</Owners>
```
Needs to be Microsoft as the owner

newky2k commented on Android/Crashlytics/source/Crashlytics/Crashlytics.csproj:
```diff
@@ -0,0 +1,74 @@
+        <Summary>Xamarin.Android bindings for Crashlytics Android</Summary>
+        <Description>Xamarin.Android bindings for Crashlytics Android</Description>
+        <PackageTags>Xamarin Android Xamarin.Android Crashlytics</PackageTags>
+        <Authors>Xamarin Inc.</Authors>
```
Needs to be Microsoft as the author

newky2k commented on Android/Crashlytics/source/Crashlytics/Crashlytics.csproj:
```diff
@@ -0,0 +1,74 @@
+        <Description>Xamarin.Android bindings for Crashlytics Android</Description>
+        <PackageTags>Xamarin Android Xamarin.Android Crashlytics</PackageTags>
+        <Authors>Xamarin Inc.</Authors>
+        <Owners>Xamarin Inc.</Owners>
```
Needs to be Microsoft as the owner

newky2k commented on Android/Crashlytics/source/CrashlyticsBeta/CrashlyticsBeta.csproj:
```diff
@@ -0,0 +1,67 @@
+        <Summary>Xamarin.Android bindings for Crashlytics Android - Beta</Summary>
+        <Description>Xamarin.Android bindings for Crashlytics Android - Beta</Description>
+        <PackageTags>Xamarin Android Xamarin.Android Crashlytics</PackageTags>
+        <Authors>Xamarin Inc.</Authors>
```
Needs to be Microsoft as the author

newky2k commented on Android/Crashlytics/source/CrashlyticsBeta/CrashlyticsBeta.csproj:
```diff
@@ -0,0 +1,67 @@
+        <Description>Xamarin.Android bindings for Crashlytics Android - Beta</Description>
+        <PackageTags>Xamarin Android Xamarin.Android Crashlytics</PackageTags>
+        <Authors>Xamarin Inc.</Authors>
+        <Owners>Xamarin Inc.</Owners>
```
Needs to be Microsoft as the owner

newky2k commented on Android/Crashlytics/source/CrashlyticsCore/CrashlyticsCore.csproj:
```diff
@@ -0,0 +1,67 @@
+        <Summary>Xamarin.Android bindings for Crashlytics Android - Core</Summary>
+        <Description>Xamarin.Android bindings for Crashlytics Android - Core</Description>
+        <PackageTags>Xamarin Android Xamarin.Android Crashlytics</PackageTags>
+        <Authors>Xamarin Inc.</Authors>
```
Needs to be Microsoft as the author

newky2k commented on Android/Crashlytics/source/CrashlyticsCore/CrashlyticsCore.csproj:
```diff
@@ -0,0 +1,67 @@
+        <Description>Xamarin.Android bindings for Crashlytics Android - Core</Description>
+        <PackageTags>Xamarin Android Xamarin.Android Crashlytics</PackageTags>
+        <Authors>Xamarin Inc.</Authors>
+        <Owners>Xamarin Inc.</Owners>
```
Needs to be Microsoft as the owner

newky2k commented on Android/Crashlytics/source/Fabric/Fabric.csproj:
```diff
@@ -0,0 +1,67 @@
+        <Summary>Xamarin.Android bindings for Fabric Android</Summary>
+        <Description>Xamarin.Android bindings for Fabric Android</Description>
+        <PackageTags>Xamarin Android Xamarin.Android Fabric</PackageTags>
+        <Authors>Xamarin Inc.</Authors>
```
Needs to be Microsoft as the author

newky2k commented on Android/Crashlytics/source/Fabric/Fabric.csproj:
```diff
@@ -0,0 +1,67 @@
+        <Description>Xamarin.Android bindings for Fabric Android</Description>
+        <PackageTags>Xamarin Android Xamarin.Android Fabric</PackageTags>
+        <Authors>Xamarin Inc.</Authors>
+        <Owners>Xamarin Inc.</Owners>
```
Needs to be Microsoft as the owners

newky2k reviewed (changes requested):
Author and Owner must be microsoft for all projects

Redth committed:
Fix authors, owner

Redth committed:
Add restore to samples

Redth committed:
Add XBD .targets to binding projects

Redth committed:
Fix task dependency chain for ‘samples’ target

Redth committed:
Set DesignTimeBuild=false

this works around a current bug where targets seem to get reordered and DesignTimeBuild isn’t set in time and is blank, causing an error later.
